subsys/llext/llext_priv.h

```diff
@@ -46,14 +51,25 @@ static inline void llext_free(void *ptr)
 int do_llext_load(struct llext_loader *ldr, struct llext *ext,
 		  struct llext_load_param *ldr_parm);
 
-elf_shdr_t *llext_section_by_name(struct llext_loader *ldr, const char *search_name);
-
 static inline const char *llext_string(struct llext_loader *ldr, struct llext *ext,
 				       enum llext_mem mem_idx, unsigned int idx)
 {
 	return (char *)ext->mem[mem_idx] + idx;
 }
 
+static inline const void *llext_elf_sect_addr(struct llext_loader *ldr, struct llext *ext,
+				unsigned int sect_idx)
```

Comment:
Nit: Not fond of the function name.
Argument name is also confusing.
Proposal:
```suggestion
static inline const void *llext_get_loaded_section_addr(struct llext_loader *ldr, struct llext *ext,
				unsigned int sh_ndx)
```

Reply:
Love the `sh_ndx`, makes it clear it is an ELF section index, we should use that more often. 

The function itself has a quirk used by the Xtensa code path: it does try to map _directly on the ELF_ via `peek` if the section is not loaded. That's why I first found your proposed name a little confusing, but that made me realize I will leave this visible where it's needed, and be unambiguous here (either a valid `ext->mem` ptr or NULL).